src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -292,12 +292,15 @@ object Ledger extends Logger {
   private def payForGasUsedToBeneficiary(stx: SignedTransaction, worldStateProxy: InMemoryWorldStateProxy): InMemoryWorldStateProxy = worldStateProxy //TODO
 
   /**
-    * Delete all accounts (that appear in SUICIDE list). YP eq (78)
+    * Delete all accounts (that appear in SUICIDE list). YP eq (78).
+    * The contract storage should be cleared during pruning as nodes could be used in other tries.
+    * The contract code is also not deleted as it could be duplicated between different contracts. FIXME: Should we keep track of this for deletion?
```

Comment:
Probably while pruning? Maybe we can update the comment to state that deleting this is risky if other account is using the same code?